pkg/scd/operational_intents_handler.go

```diff
@@ -641,40 +689,15 @@ func (a *Server) upsertOperationalIntentReference(ctx context.Context, authorize
 
 		var sub *scdmodels.Subscription
 		if validParams.subscriptionID.Empty() {
-			// Create an implicit subscription if the implicit subscription params are set:
-			// for situations where these params are required but have not been set,
-			// an error will have been returned earlier.
-			// If they are not set at this point, continue without creating an implicit subscription.
-			if params.NewSubscription != nil && params.NewSubscription.UssBaseUrl != "" {
-				if !a.AllowHTTPBaseUrls {
-					err := scdmodels.ValidateUSSBaseURL(string(params.NewSubscription.UssBaseUrl))
-					if err != nil {
-						return stacktrace.PropagateWithCode(err, dsserr.BadRequest, "Failed to validate USS base URL")
-					}
-				}
-
-				subToUpsert := scdmodels.Subscription{
-					ID:                          dssmodels.ID(uuid.New().String()),
-					Manager:                     manager,
-					StartTime:                   validParams.uExtent.StartTime,
-					EndTime:                     validParams.uExtent.EndTime,
-					AltitudeLo:                  validParams.uExtent.SpatialVolume.AltitudeLo,
-					AltitudeHi:                  validParams.uExtent.SpatialVolume.AltitudeHi,
-					Cells:                       validParams.cells,
-					USSBaseURL:                  string(params.NewSubscription.UssBaseUrl),
-					NotifyForOperationalIntents: true,
-					ImplicitSubscription:        true,
-				}
-				if params.NewSubscription.NotifyForConstraints != nil {
-					subToUpsert.NotifyForConstraints = *params.NewSubscription.NotifyForConstraints
-				}
-
-				sub, err = r.UpsertSubscription(ctx, &subToUpsert)
-				if err != nil {
+			// Create an implicit subscription if one has been requested.
+			// Requesting neither an explicit nor an implicit subscription is allowed for ACCEPTED states:
+			// for other states, an error will have been returned earlier.
+			// if no implicit subscription is requested and we reached this point, we will proceed without subscription
+			if validParams.implicitSubscriptionRequested {
```

Comment:
Actually now we don't need anymore to check for `validParams.subscriptionID.Empty()` above: we can directly check for `validParams.implicitSubscriptionRequested`.

Reply:
Unfortunately not:

`validParams.subscriptionID.Empty()` is allowed to be empty while `validParams.implicitSubscriptionRequested` is false when the OIR is in the `ACCEPTED` state.

When we enter the `validParams.subscriptionID.Empty()` branch, we only want to do something if `validParams.implicitSubscriptionRequested` is true.

If we do `validParams.implicitSubscriptionRequested` in the place where we currently check `validParams.subscriptionID.Empty()`, we will go through the `// Use existing Subscription` branch.

There's probably a better way to express this (possibly with a switch), but I'll be keeping that for later.

Reply:
Right! Indeed the different is subtle.